### PR TITLE
Add sticky bottom utility

### LINKED ISSUE #35503: sticky-${breakpoint}-bottom
### Prerequisites

- [X] I have [searched](https://github.com/twbs/bootstrap/issues?utf8=%E2%9C%93&q=is%3Aissue) for duplicate or closed feature requests
- [X] I have read the [contributing guidelines](https://github.com/twbs/bootstrap/blob/main/.github/CONTRIBUTING.md)

### Proposal

With .sticky-top we have a custom z-index value and responsive variations (e.g. .sticky-lg-top). It would be convenient to have the same utilities for a bottom sticky, like .sticky-${breakpoint}-bottom.

### PR DESCRIPTION
**Pull request by florianlacreuse**
Fixes #35503.

### CODE REVIEW

**Comment by WinterSilence** [comment]
how it's work?

**Comment by florianlacreuse** [comment]
Live example: https://codepen.io/florianlacreuse/pen/wvrzdpY

**Comment by WinterSilence** [comment]
@florianlacreuse
> Note that a sticky element "sticks" to its nearest ancestor that has a "scrolling mechanism" (created when overflow is hidden, scroll, auto, or overlay), even if that ancestor isn't the nearest actually scrolling ancestor

i.e. it's work fine if page have long body, in other case your `sticky-bottom` footer displayed at center page.

**Comment by florianlacreuse** [comment]
I'm not sure what you mean. An element with `.sticky-top` can also be in middle of the page.

`.sticky-top` and `.sticky-bottom` are different than "sticky header" and "sticky footer". See: https://getbootstrap.com/docs/5.1/examples/sticky-footer/

**Comment by WinterSilence** [comment]
https://codepen.io/WinterSilence/pen/BawLddW

**Comment by WinterSilence** [comment]
@florianlacreuse 
> .sticky-top and .sticky-bottom are different than "sticky header" and "sticky footer". 
see [sticky header in manual](https://getbootstrap.com/docs/5.1/components/navbar/#placement)
users 100% will try add `.sticky-bottom` to footer - need notice displaying when sticky doesn't work as expected.

**Comment by florianlacreuse** [comment]
Thanks for the live example, I understand better what you meant.

Yes, `.sticky-bottom` will not magically position the target element at the bottom of its "container" / ancestor. And neither does the `.sticky-top`, just that by default the elements are usually "on top".

Your example fixed: https://codepen.io/florianlacreuse/pen/MWEjvPP

**Comment by WinterSilence** [comment]
@florianlacreuse yes, add this comment to `position.md`

**Review by ffoodd — CHANGES_REQUESTED** [review]
Thanks for the PR!

LGTM, and after double checking how `position: sticky` support compares to our current browser target, I confirm we can drop sentences regarding browser support.

**Comment by ffoodd** [comment]
Regarding edge cases, I don't think we should mention them. This is how `position` works, and is really easy to mess up when you don't understand containing block.

For example, `position: fixed` does not always refer to the viewport since it refers to its containing block. Any element between the fixed element and the viewport, with e.g. a `transform`, will become its containing block. 

We don't mention such edge cases as it's in the CSS specification and we don't aim at proxying it. At least IMHO, this is still open to discussions obviously 🙂

File: site/content/docs/5.1/components/navbar.md [hunk 1]
```diff
@@ -450,9 +450,11 @@ Use any of the responsive containers to change how wide the content in your navb
...
+Use our [position utilities]({{< docsref "/utilities/position" >}}) to place navbars in non-static positions. Choose from fixed to the top, fixed to the bottom, stickied to the top (scrolls with the page until it reaches the top, then stays there), or stickied to the bottom (scrolls with the page until it reaches the bottom, then stays there).
 
-Also note that **`.sticky-top` uses `position: sticky`, which [isn't fully supported in every browser](https://caniuse.com/css-sticky)**.
+Fixed navbars use `position: fixed`, meaning they're pulled from the normal flow of the DOM and may require custom CSS (e.g., `padding-top` on the `<body>`) to prevent overlap with other elements.
+
+Also note that **`.sticky-top` and `.sticky-bottom` uses `position: sticky`, which [isn't fully supported in every browser](https://caniuse.com/css-sticky)**.
```
[the remark this answers is not in the document]
Comment (florianlacreuse) [reply site/content/docs/5.1/components/navbar.md]: You are right about that, I dropped it.

File: site/content/docs/5.1/helpers/position.md [hunk 2]
```diff
@@ -40,3 +40,22 @@ Responsive variations also exist for `.sticky-top` utility.
 <div class="sticky-lg-top">Stick to the top on viewports sized LG (large) or wider</div>
 <div class="sticky-xl-top">Stick to the top on viewports sized XL (extra-large) or wider</div>
 ```
+
+## Sticky bottom
+
+Position an element at the bottom of the viewport, from edge to edge, but only after you scroll past it. The `.sticky-bottom` utility uses CSS's `position: sticky`, which isn't fully supported in all browsers.
```
[the remark this answers is not in the document]
Comment (florianlacreuse) [reply site/content/docs/5.1/helpers/position.md]: Done.

**Comment by florianlacreuse** [comment]
I agree with @ffoodd, this is pure CSS specs.

I dropped the sentences about browser support.

**Review by ffoodd — APPROVED** [review]
Great, thanks!

**Pull request merged**